homeassistant/components/nws/sensor.py

```diff
@@ -39,65 +39,59 @@ async def async_setup_entry(hass, entry, async_add_entities):
     hass_data = hass.data[DOMAIN][entry.entry_id]
     station = entry.data[CONF_STATION]
 
-    entities = []
-    for sensor_type, metadata in SENSOR_TYPES.items():
-        entities.append(
-            NWSSensor(
-                hass,
-                entry.data,
-                hass_data,
-                sensor_type,
-                metadata,
-                station,
-            ),
+    async_add_entities(
+        NWSSensor(
+            hass=hass,
+            entry_data=entry.data,
+            hass_data=hass_data,
+            description=description,
+            station=station,
         )
-
-    async_add_entities(entities, False)
+        for description in SENSOR_TYPES
+    )
 
 
 class NWSSensor(CoordinatorEntity, SensorEntity):
     """An NWS Sensor Entity."""
 
+    entity_description: NWSSensorEntityDescription
+
     def __init__(
         self,
         hass: HomeAssistant,
         entry_data,
         hass_data,
-        sensor_type,
-        metadata: NWSSensorMetadata,
+        description: NWSSensorEntityDescription,
         station,
     ):
         """Initialise the platform with a data instance."""
         super().__init__(hass_data[COORDINATOR_OBSERVATION])
         self._nws = hass_data[NWS_DATA]
         self._latitude = entry_data[CONF_LATITUDE]
         self._longitude = entry_data[CONF_LONGITUDE]
-        self._type = sensor_type
-        self._metadata = metadata
+        self.entity_description = description
 
-        self._attr_name = f"{station} {metadata.label}"
-        self._attr_icon = metadata.icon
-        self._attr_device_class = metadata.device_class
-        if hass.config.units.is_metric:
-            self._attr_unit_of_measurement = metadata.unit
-        else:
-            self._attr_unit_of_measurement = metadata.unit_convert
+        self._attr_name = f"{station} {description.name}"
+        if not hass.config.units.is_metric:
+            self._attr_unit_of_measurement = description.unit_convert
 
     @property
     def state(self):
         """Return the state."""
-        value = self._nws.observation.get(self._type)
+        value = self._nws.observation.get(self.entity_description.key)
         if value is None:
             return None
-        if self._attr_unit_of_measurement == SPEED_MILES_PER_HOUR:
+        # Set alias to unit property -> prevent unnecessary hasattr calls
+        unit_of_measurement = self.unit_of_measurement
+        if unit_of_measurement == SPEED_MILES_PER_HOUR:
```

Comment:
@frenck Not sure if that's the best way to go about about it.
(True, this entity should probably be refactored so this isn't an issue but that is not the objective of this PR.)

Reply:
I think either way is fine, either by accessing the property or in this case the entity_description could have been used directly as well.